src/api/Interpret.h

```diff
@@ -64,6 +64,55 @@ class DefinedFunctions {
     }
 };
 
+class TermNames {
+public:
+    bool has(std::string const & name) const { return nameToTerm.find(name) != nameToTerm.end(); }
+    bool has(PTRef term) const { return termToNames.find(term) != termToNames.end(); }
+
+    void insert(std::string const & name, PTRef term, bool scoped = false) {
+        assert(not has(name));
+        nameToTerm[name] = term;
+        termToNames[term].push_back(name);
+        if (scoped) { scopedNames.push(name); }
+    }
+
+    void pushScope() { scopedNames.pushScope(); }
+
+    void popScope() {
+        scopedNames.popScope([&](std::string const & name) {
+            auto it = nameToTerm.find(name);
+            if (it != nameToTerm.end()) {
```

Comment:
`if (it == nameToTerm.end()) return;` avoids adding another scope and indentation

Reply:
Fixed!